dev-lang/vala/vala-0.40.3.recipe

```diff
@@ -0,0 +1,106 @@
+SUMMARY="An object-oriented programming language that uses the GObject system"
+DESCRIPTION="Vala is a new programming language that aims to bring modern \
+programming language features to GNOME developers without imposing any \
+additional runtime requirements and without using a different ABI \
+compared to applications and libraries written in C."
+HOMEPAGE="https://wiki.gnome.org/Projects/Vala"
+COPYRIGHT="2006-2018 Jürg Billeter
+	2006-2018 Raffaele Sandrini"
+LICENSE="GNU LGPL v2.1"
+REVISION="1"
+SOURCE_URI="https://download.gnome.org/sources/vala/${portVersion%.*}/vala-$portVersion.tar.xz"
+CHECKSUM_SHA256="6d4f1f0b2edaa0d4aa96f72735a9845e6b1acf80a0a2ae494b5d43e07833119e"
+SOURCE_DIR="vala-$portVersion"
+PATCHES="vala-$portVersion.patchset"
+
+ARCHITECTURES="!x86_gcc2 ?x86 ?x86_64"
+SECONDARY_ARCHITECTURES="?x86"
```

Comment:
```diff
-ARCHITECTURES="!x86_gcc2 ?x86 ?x86_64"
-SECONDARY_ARCHITECTURES="?x86"
+ARCHITECTURES="!x86_gcc2 x86 x86_64"
+SECONDARY_ARCHITECTURES="x86"
```
I have tested vala 0.40.3 on all archictectures, (including x86 primary), and all tests are OK:<details>
```
make[4]: Entering directory '/sources/vala-0.40.3/tests'
TEST: Building...                                                      OK
...
All 474 tests passed
TEST: Building...                                                      OK
  /nullability/bug611223:                                              OK
All 1 tests passed
...
make[6]: Entering directory '/sources/vala-0.40.3/valadoc/tests'
TEST: Building...                                                      OK
...
All 8 tests passed
```

BTW, the build failures I had previously on x86_64 were due to a buggy build of graphviz.
After I installed `graphviz{,_devel}` from the repo I was able to successfully build vala and pass all the tests.
</details>

Reply:
I think libintl isn't needed too, autoreconf doesn't use it (not using gettext) ...

Reply:
I don't know if libintl is required for the build to succeed, but I can say that the shared libs we obtain (when we build vala with `devel:libintl$secondaryArchSuffix` in `BUILD_REQUIRES`) all need libintl:
```bash
cd /system
for i in $(package list -p /system/packages/vala-0.40.3-1-x86_64.hpkg | grep '\.so$' ) ; do objdump -x $i | grep intl && echo $i; done
  NEEDED               libintl.so.8
develop/lib/libvala-0.40.so
  NEEDED               libintl.so.8
develop/lib/libvaladoc-0.40.so
  NEEDED               libintl.so.8
lib/vala-0.40/libvalaccodegen.so
  NEEDED               libintl.so.8
lib/valadoc/doclets/devhelp/libdoclet.so
  NEEDED               libintl.so.8
lib/valadoc/doclets/gtkdoc/libdoclet.so
  NEEDED               libintl.so.8
lib/valadoc/doclets/html/libdoclet.so
```

Reply:
In this PR 0.36.12 wasn't in my branch, so I can't remove it from this branch ...

Reply:
No problem. After all, that's a good idea to keep a recipe for vala `0.36.x` because that's a LTS branch.

Reply:
Ok, so if this is ok could this be merged, or do you still have problems building it?

Reply:
I did not succeed yet, but I did not investigate much...